### PR TITLE
Fix ImplicitMachineConfigHost parameter checking

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Don't assume index of the ConfigurationFileMap in params

Fixes: #28784

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
> Looks like it will fix the problem. Thanks.

I did repro the crash with the test before fixing it so I'm pretty confident. If you hit anything else please let me know.

File: src/System.Configuration.ConfigurationManager/src/System/Configuration/ImplicitMachineConfigHost.cs [hunk 1]
```diff
@@ -23,17 +23,27 @@ internal ImplicitMachineConfigHost(IInternalConfigHost host)
...
         {
             // Stash the filemap so we can see if the machine config was explicitly specified
-            _fileMap = (ConfigurationFileMap)hostInitParams[0];
+            GetFileMap(hostInitParams);
             base.Init(configRoot, hostInitParams);
         }
 
+        private void GetFileMap(object[] parameters)
+        {
+            foreach (object parameter in parameters)
```
Comment (stephentoub) [inline src/System.Configuration.ConfigurationManager/src/System/Configuration/ImplicitMachineConfigHost.cs]: Is it possible parameter could be null and we should early out on that, too?

Comment (JeremyKuhne) [reply src/System.Configuration.ConfigurationManager/src/System/Configuration/ImplicitMachineConfigHost.cs]: We don't check in the various host implementations so I didn't go that far.

**Comment by JeremyKuhne** [comment]
Hmm. Looks like we have an error in the Console title code on NetFX.

```
Unhandled Exception of Type System.IO.IOException
Message :
System.IO.IOException : The data area passed to a system call is too small.
Stack Trace :
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.Console.get_Title()
   at WindowAndCursorProps.Title_Get_Windows() in D:\j\workspace\windows-TGrou---2a8f9c29\src\System.Console\tests\WindowAndCursorProps.cs:line 139
```

That is an `ERROR_INSUFFICIENT_BUFFER`. Looking at the CoreFX code I think we might have that wrong too. Investigating.

(Clearly this has nothing to do with my change.)

**Comment by danmosemsft** [comment]
@JeremyKuhne recall taht was https://github.com/dotnet/corefx/pull/21518 I suggest you ignore it.

**Comment by JeremyKuhne** [comment]
> recall taht was #21518 I suggest you ignore it. Although that would be odd because it was fixed in RS3 - the CI should be updated apparently

It might be. That said it seems pretty clear that the bug fell over on the size we actually put in the buffer ((256 + 1) * 2). The code here isn't great, I've already cleaned it up and tried to allow for the RS2 issue. I'll put up a PR for post 2.1.

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Linux x64 Release Build
@dotnet-bot test NETFX x86 Release Build

**Comment by JeremyKuhne** [comment]
Saving (which we weren't doing in the tests) isn't working on the Unix CI machines (the filename is empty for some reason). I can't repro it locally- still investigating.

**Comment by stephentoub** [comment]
@JeremyKuhne, does this need to get in for 2.1?

**Comment by JeremyKuhne** [comment]
> does this need to get in for 2.1?

Moving to 2.2 for now. We've been broken this way from the start. I need to figure out why we're only failing on the Linux x64 CI machines with this change in place.

**Comment by stephentoub** [comment]
> I need to figure out why we're only failing on the Linux x64 CI machines with this change in place.

@JeremyKuhne, it's been two months... still working on this?  Should the PR be closed in the interim?  The change somehow special-cased to Windows? Etc.

**Comment by JeremyKuhne** [comment]
@dotnet-bot test this please

**Comment by JeremyKuhne** [comment]
> still working on this?

Kicking a run again to see if the same weirdness is still happening. Will follow up based on results.

**Comment by stephentoub** [comment]
Failing on Nano with:
```
2018-06-12 18:18:19,084: INFO: proc(54): run_and_log_output: Output:       System.Configuration.ConfigurationErrorsException : Failed to save settings: An error occurred loading a configuration file: An unexpected error occurred in 'ClientConfigurationHost::OpenStreamForWrite '' '''.
2018-06-12 18:18:19,085: INFO: proc(54): run_and_log_output: Output:       ---- System.Configuration.ConfigurationErrorsException : An error occurred loading a configuration file: An unexpected error occurred in 'ClientConfigurationHost::OpenStreamForWrite '' '''.
2018-06-12 18:18:19,085: INFO: proc(54): run_and_log_output: Output:       -------- System.InvalidOperationException : An unexpected error occurred in 'ClientConfigurationHost::OpenStreamForWrite '' '''.
2018-06-12 18:18:19,091: INFO: proc(54): run_and_log_output: Output:       Stack Trace:
2018-06-12 18:18:19,125: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\ClientSettingsStore.cs(209,0): at System.Configuration.ClientSettingsStore.WriteSettings(String sectionName, Boolean isRoaming, IDictionary newSettings)
2018-06-12 18:18:19,125: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\LocalFileSettingsProvider.cs(218,0): at System.Configuration.LocalFileSettingsProvider.SetPropertyValues(SettingsContext context, SettingsPropertyValueCollection values)
2018-06-12 18:18:19,127: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\SettingsBase.cs(141,0): at System.Configuration.SettingsBase.SaveCore()
2018-06-12 18:18:19,127: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\SettingsBase.cs(115,0): at System.Configuration.SettingsBase.Save()
2018-06-12 18:18:19,128: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\ApplicationSettingsBase.cs(389,0): at System.Configuration.ApplicationSettingsBase.Save()
2018-06-12 18:18:19,128: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\tests\System\Configuration\ApplicationSettingsBaseTests.cs(132,0): at System.ConfigurationTests.ApplicationSettingsBaseTests.Save_SimpleSettings_Ok(Boolean isSynchronized)
2018-06-12 18:18:19,128: INFO: proc(54): run_and_log_output: Output:          ----- Inner Stack Trace -----
2018-06-12 18:18:19,128: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\MgmtConfigurationRecord.cs(1098,0): at System.Configuration.MgmtConfigurationRecord.SaveAs(String filename, ConfigurationSaveMode saveMode, Boolean forceUpdateAll)
2018-06-12 18:18:19,130: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\Configuration.cs(217,0): at System.Configuration.Configuration.SaveAsImpl(String filename, ConfigurationSaveMode saveMode, Boolean forceSaveAll)
2018-06-12 18:18:19,130: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\Configuration.cs(182,0): at System.Configuration.Configuration.Save()
2018-06-12 18:18:19,131: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\ClientSettingsStore.cs(204,0): at System.Configuration.ClientSettingsStore.WriteSettings(String sectionName, Boolean isRoaming, IDictionary newSettings)
2018-06-12 18:18:19,131: INFO: proc(54): run_and_log_output: Output:          ----- Inner Stack Trace -----
2018-06-12 18:18:19,131: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\ClientConfigurationHost.cs(314,0): at System.Configuration.ClientConfigurationHost.OpenStreamForWrite(String streamName, String templateStreamName, Object& writeContext)
2018-06-12 18:18:19,131: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\ClientSettingsStore.cs(327,0): at System.Configuration.ClientSettingsStore.ClientSettingsConfigurationHost.OpenStreamForWrite(String streamName, String templateStreamName, Object& writeContext)
2018-06-12 18:18:19,131: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\UpdateConfigHost.cs(94,0): at System.Configuration.UpdateConfigHost.OpenStreamForWrite(String streamName, String templateStreamName, Object& writeContext)
2018-06-12 18:18:19,132: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\Internal\DelegatingConfigHost.cs(96,0): at System.Configuration.Internal.DelegatingConfigHost.OpenStreamForWrite(String streamName, String templateStreamName, Object& writeContext)
2018-06-12 18:18:19,132: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\MgmtConfigurationRecord.cs(1062,0): at System.Configuration.MgmtConfigurationRecord.SaveAs(String filename, ConfigurationSaveMode saveMode, Boolean forceUpdateAll)
2018-06-12 18:18:19,134: INFO: proc(54): run_and_log_output: Output:    System.ConfigurationTests.ApplicationSettingsBaseTests.Save_SimpleSettings_Ok(isSynchronized: False) [FAIL]
2018-06-12 18:18:19,134: INFO: proc(54): run_and_log_output: Output:       System.Configuration.ConfigurationErrorsException : Failed to save settings: An error occurred loading a configuration file: An unexpected error occurred in 'ClientConfigurationHost::OpenStreamForWrite '' '''.
2018-06-12 18:18:19,135: INFO: proc(54): run_and_log_output: Output:       ---- System.Configuration.ConfigurationErrorsException : An error occurred loading a configuration file: An unexpected error occurred in 'ClientConfigurationHost::OpenStreamForWrite '' '''.
2018-06-12 18:18:19,135: INFO: proc(54): run_and_log_output: Output:       -------- System.InvalidOperationException : An unexpected error occurred in 'ClientConfigurationHost::OpenStreamForWrite '' '''.
2018-06-12 18:18:19,135: INFO: proc(54): run_and_log_output: Output:       Stack Trace:
2018-06-12 18:18:19,137: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\ClientSettingsStore.cs(209,0): at System.Configuration.ClientSettingsStore.WriteSettings(String sectionName, Boolean isRoaming, IDictionary newSettings)
2018-06-12 18:18:19,138: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\LocalFileSettingsProvider.cs(218,0): at System.Configuration.LocalFileSettingsProvider.SetPropertyValues(SettingsContext context, SettingsPropertyValueCollection values)
2018-06-12 18:18:19,138: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\SettingsBase.cs(141,0): at System.Configuration.SettingsBase.SaveCore()
2018-06-12 18:18:19,140: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\SettingsBase.cs(120,0): at System.Configuration.SettingsBase.Save()
2018-06-12 18:18:19,140: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\ApplicationSettingsBase.cs(389,0): at System.Configuration.ApplicationSettingsBase.Save()
2018-06-12 18:18:19,140: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\tests\System\Configuration\ApplicationSettingsBaseTests.cs(132,0): at System.ConfigurationTests.ApplicationSettingsBaseTests.Save_SimpleSettings_Ok(Boolean isSynchronized)
2018-06-12 18:18:19,141: INFO: proc(54): run_and_log_output: Output:          ----- Inner Stack Trace -----
2018-06-12 18:18:19,141: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\MgmtConfigurationRecord.cs(1098,0): at System.Configuration.MgmtConfigurationRecord.SaveAs(String filename, ConfigurationSaveMode saveMode, Boolean forceUpdateAll)
2018-06-12 18:18:19,141: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\Configuration.cs(217,0): at System.Configuration.Configuration.SaveAsImpl(String filename, ConfigurationSaveMode saveMode, Boolean forceSaveAll)
2018-06-12 18:18:19,141: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\Configuration.cs(182,0): at System.Configuration.Configuration.Save()
2018-06-12 18:18:19,142: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\ClientSettingsStore.cs(204,0): at System.Configuration.ClientSettingsStore.WriteSettings(String sectionName, Boolean isRoaming, IDictionary newSettings)
2018-06-12 18:18:19,142: INFO: proc(54): run_and_log_output: Output:          ----- Inner Stack Trace -----
2018-06-12 18:18:19,142: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\ClientConfigurationHost.cs(314,0): at System.Configuration.ClientConfigurationHost.OpenStreamForWrite(String streamName, String templateStreamName, Object& writeContext)
2018-06-12 18:18:19,144: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\ClientSettingsStore.cs(327,0): at System.Configuration.ClientSettingsStore.ClientSettingsConfigurationHost.OpenStreamForWrite(String streamName, String templateStreamName, Object& writeContext)
2018-06-12 18:18:19,144: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\UpdateConfigHost.cs(94,0): at System.Configuration.UpdateConfigHost.OpenStreamForWrite(String streamName, String templateStreamName, Object& writeContext)
2018-06-12 18:18:19,144: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\Internal\DelegatingConfigHost.cs(96,0): at System.Configuration.Internal.DelegatingConfigHost.OpenStreamForWrite(String streamName, String templateStreamName, Object& writeContext)
2018-06-12 18:18:19,144: INFO: proc(54): run_and_log_output: Output:          D:\j\workspace\windows-TGrou---74aa877a\src\System.Configuration.ConfigurationManager\src\System\Configuration\MgmtConfigurationRecord.cs(1062,0): at System.Configuration.MgmtConfigurationRecord.SaveAs(String filename, ConfigurationSaveMode saveMode, Boolean forceUpdateAll)
```

Failing on Linux with:
```
2018-06-12 18:36:32,792: INFO: proc(54): run_and_log_output: Output:    System.ConfigurationTests.ApplicationSettingsBaseTests.Save_SimpleSettings_Ok(isSynchronized: True) [FAIL]
2018-06-12 18:36:32,794: INFO: proc(54): run_and_log_output: Output:       System.Configuration.ConfigurationErrorsException : Failed to save settings: An error occurred loading a configuration file: An unexpected error occurred in 'ClientConfigurationHost::OpenStreamForWrite '' '''.
2018-06-12 18:36:32,795: INFO: proc(54): run_and_log_output: Output:       ---- System.Configuration.ConfigurationErrorsException : An error occurred loading a configuration file: An unexpected error occurred in 'ClientConfigurationHost::OpenStreamForWrite '' '''.
2018-06-12 18:36:32,795: INFO: proc(54): run_and_log_output: Output:       -------- System.InvalidOperationException : An unexpected error occurred in 'ClientConfigurationHost::OpenStreamForWrite '' '''.
2018-06-12 18:36:32,803: INFO: proc(54): run_and_log_output: Output:       Stack Trace:
2018-06-12 18:36:32,836: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/src/System/Configuration/ClientSettingsStore.cs(209,0): at System.Configuration.ClientSettingsStore.WriteSettings(String sectionName, Boolean isRoaming, IDictionary newSettings)
2018-06-12 18:36:32,837: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/src/System/Configuration/LocalFileSettingsProvider.cs(220,0): at System.Configuration.LocalFileSettingsProvider.SetPropertyValues(SettingsContext context, SettingsPropertyValueCollection values)
2018-06-12 18:36:32,837: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/src/System/Configuration/SettingsBase.cs(129,0): at System.Configuration.SettingsBase.SaveCore()
2018-06-12 18:36:32,838: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/src/System/Configuration/SettingsBase.cs(115,0): at System.Configuration.SettingsBase.Save()
2018-06-12 18:36:32,838: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/tests/System/Configuration/ApplicationSettingsBaseTests.cs(135,0): at System.ConfigurationTests.ApplicationSettingsBaseTests.Save_SimpleSettings_Ok(Boolean isSynchronized)
2018-06-12 18:36:32,838: INFO: proc(54): run_and_log_output: Output:          ----- Inner Stack Trace -----
2018-06-12 18:36:32,840: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/src/System/Configuration/MgmtConfigurationRecord.cs(1098,0): at System.Configuration.MgmtConfigurationRecord.SaveAs(String filename, ConfigurationSaveMode saveMode, Boolean forceUpdateAll)
2018-06-12 18:36:32,840: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/src/System/Configuration/ClientSettingsStore.cs(204,0): at System.Configuration.ClientSettingsStore.WriteSettings(String sectionName, Boolean isRoaming, IDictionary newSettings)
2018-06-12 18:36:32,840: INFO: proc(54): run_and_log_output: Output:          ----- Inner Stack Trace -----
2018-06-12 18:36:32,840: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigurationHost.cs(312,0): at System.Configuration.ClientConfigurationHost.OpenStreamForWrite(String streamName, String templateStreamName, Object& writeContext)
2018-06-12 18:36:32,841: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/src/System/Configuration/MgmtConfigurationRecord.cs(1062,0): at System.Configuration.MgmtConfigurationRecord.SaveAs(String filename, ConfigurationSaveMode saveMode, Boolean forceUpdateAll)
2018-06-12 18:36:32,843: INFO: proc(54): run_and_log_output: Output:    System.ConfigurationTests.ApplicationSettingsBaseTests.Save_SimpleSettings_Ok(isSynchronized: False) [FAIL]
2018-06-12 18:36:32,843: INFO: proc(54): run_and_log_output: Output:       System.Configuration.ConfigurationErrorsException : Failed to save settings: An error occurred loading a configuration file: An unexpected error occurred in 'ClientConfigurationHost::OpenStreamForWrite '' '''.
2018-06-12 18:36:32,843: INFO: proc(54): run_and_log_output: Output:       ---- System.Configuration.ConfigurationErrorsException : An error occurred loading a configuration file: An unexpected error occurred in 'ClientConfigurationHost::OpenStreamForWrite '' '''.
2018-06-12 18:36:32,843: INFO: proc(54): run_and_log_output: Output:       -------- System.InvalidOperationException : An unexpected error occurred in 'ClientConfigurationHost::OpenStreamForWrite '' '''.
2018-06-12 18:36:32,844: INFO: proc(54): run_and_log_output: Output:       Stack Trace:
2018-06-12 18:36:32,844: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/src/System/Configuration/ClientSettingsStore.cs(209,0): at System.Configuration.ClientSettingsStore.WriteSettings(String sectionName, Boolean isRoaming, IDictionary newSettings)
2018-06-12 18:36:32,844: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/src/System/Configuration/LocalFileSettingsProvider.cs(220,0): at System.Configuration.LocalFileSettingsProvider.SetPropertyValues(SettingsContext context, SettingsPropertyValueCollection values)
2018-06-12 18:36:32,844: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/src/System/Configuration/SettingsBase.cs(129,0): at System.Configuration.SettingsBase.SaveCore()
2018-06-12 18:36:32,849: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/src/System/Configuration/SettingsBase.cs(122,0): at System.Configuration.SettingsBase.Save()
2018-06-12 18:36:32,849: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/tests/System/Configuration/ApplicationSettingsBaseTests.cs(135,0): at System.ConfigurationTests.ApplicationSettingsBaseTests.Save_SimpleSettings_Ok(Boolean isSynchronized)
2018-06-12 18:36:32,849: INFO: proc(54): run_and_log_output: Output:          ----- Inner Stack Trace -----
2018-06-12 18:36:32,850: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/src/System/Configuration/MgmtConfigurationRecord.cs(1098,0): at System.Configuration.MgmtConfigurationRecord.SaveAs(String filename, ConfigurationSaveMode saveMode, Boolean forceUpdateAll)
2018-06-12 18:36:32,850: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/src/System/Configuration/ClientSettingsStore.cs(204,0): at System.Configuration.ClientSettingsStore.WriteSettings(String sectionName, Boolean isRoaming, IDictionary newSettings)
2018-06-12 18:36:32,850: INFO: proc(54): run_and_log_output: Output:          ----- Inner Stack Trace -----
2018-06-12 18:36:32,851: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigurationHost.cs(312,0): at System.Configuration.ClientConfigurationHost.OpenStreamForWrite(String streamName, String templateStreamName, Object& writeContext)
2018-06-12 18:36:32,851: INFO: proc(54): run_and_log_output: Output:          /mnt/j/workspace/dotnet_corefx/master/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/src/System.Configuration.ConfigurationManager/src/System/Configuration/MgmtConfigurationRecord.cs(1062,0): at System.Configuration.MgmtConfigurationRecord.SaveAs(String filename, ConfigurationSaveMode saveMode, Boolean forceUpdateAll)
```

**Comment by JeremyKuhne** [comment]
Thanks @stephentoub! I'll investigate today and update.

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Linux x64 Release Build
@dotnet-bot test Windows x64 Debug Build

**Comment by karelz** [comment]
@JeremyKuhne this is rather old PR. If it is not ready for merge, let's close it until you have time to finish it off. Thanks!

**Comment by JeremyKuhne** [comment]
Closing to avoid distraction

**Comment by maryamariyan** [comment]
push forced a rebase to resolve conflict. will wait for the CI build to finish to check for remaining todo to close this PR (To fix, skip, or remove remaining failing tests).

**Comment by maryamariyan** [comment]
@dotnet-bot test Linux x64 Debug Build
@dotnet-bot test Linux x64 Release Build

**Comment by maryamariyan** [comment]
I'm not getting a repro locally on Linux 16.04 on either Debug or Release

**Comment by maryamariyan** [comment]
I tried the repro tool and it seems broken. It seems there's an open issue for it. 

provided more details offline to @alexperovich @MattGal @Anipik

**Comment by maryamariyan** [comment]
@dotnet-bot test this please

**Comment by maryamariyan** [comment]
I plan to create a separate issue for the Linux failures and merge this PR which contains tests that pass on windows.
The Linux issue may be something weird about how our machines are configured.

File: src/System.Configuration.ConfigurationManager/tests/Helpers.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+using System.ComponentModel;
+using System.Configuration;
+using Xunit;
+
+namespace System.ConfigurationTests
+{
+    public static class Helpers
+    {
+        public const string IsSaveSettingsSupported = nameof(Helpers) + "." + nameof(GetIsSaveSettingsSupported);
```
Comment (JeremyKuhne) [inline src/System.Configuration.ConfigurationManager/tests/Helpers.cs]: There is no point in adding two if only one is used. That said, wouldn't it be more appropriate to put another `ActiveIssue` tag on the method as you intend to follow up?

**Comment by maryamariyan** [comment]
looking at the logs I see these lines (basically mentions that the job failed.)

> 2019-02-28T18:10:09.6361024Z C:\Users\VssAdministrator\.nuget\packages\microsoft.dotnet.helix.sdk\2.0.0-beta.19126.3\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(67,5): error : Work item System.Configuration.ConfigurationManager.Tests in job 4380d4a3-0c38-4c03-bbdf-d0d33876d509 has Passed with exit code 1 [D:\a\1\s\eng\sendtohelix.proj]
2019-02-28T18:10:24.1976878Z C:\Users\VssAdministrator\.nuget\packages\microsoft.dotnet.helix.sdk\2.0.0-beta.19126.3\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(67,5): error : Job '4380d4a3-0c38-4c03-bbdf-d0d33876d509' failed 2 out of 383045 tests. [D:\a\1\s\eng\sendtohelix.proj]
2019-02-28T18:10:24.2076548Z
2019-02-28T18:10:24.2076866Z Build FAILED.
2019-02-28T18:10:24.2077384Z
2019-02-28T18:10:24.2077750Z C:\Users\VssAdministrator\.nuget\packages\microsoft.dotnet.helix.sdk\2.0.0-beta.19126.3\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(67,5): error : Work item System.Configuration.ConfigurationManager.Tests in job 4380d4a3-0c38-4c03-bbdf-d0d33876d509 has Passed with exit code 1 [D:\a\1\s\eng\sendtohelix.proj]
2019-02-28T18:10:24.2078324Z C:\Users\VssAdministrator\.nuget\packages\microsoft.dotnet.helix.sdk\2.0.0-beta.19126.3\tools\Microsoft.DotNet.Helix.Sdk.MultiQueue.targets(67,5): error : Job '4380d4a3-0c38-4c03-bbdf-d0d33876d509' failed 2 out of 383045 tests. [D:\a\1\s\eng\sendtohelix.proj]
2019-02-28T18:10:24.2078579Z     0 Warning(s)
2019-02-28T18:10:24.2079107Z     2 Error(s)

@safern I couldn't find more info than that. In case there is somewhere else I am missing please let me know.

File: src/System.Configuration.ConfigurationManager/tests/Helpers.cs (hunk 2 above)
Comment (maryamariyan) [reply src/System.Configuration.ConfigurationManager/tests/Helpers.cs]: Good catch. I didn't realized the other one was added by you in the same PR, thought rather existed also for other tests.

**Comment by safern** [comment]
Under the "Send to Helix" logs there is a line pointing to mission control URL: 
`Results will be available from https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F28894~2Fmerge/test~2Ffunctional~2Fcli~2F/20190228.11
  Sent Helix Job 50852533-408f-4229-b138-1123477fc07a
`

This should improve once we move to Azure DevOps test explorer

**Comment by maryamariyan** [comment]
@JeremyKuhne @danmosemsft Could you please do a final review?

File: src/System.Configuration.ConfigurationManager/tests/System/Configuration/ApplicationSettingsBaseTests.cs [hunk 3]
```diff
@@ -103,12 +107,59 @@ public void GetSetIntProperty_SimpleSettings_Ok(bool isSynchronized)
             Assert.Equal(10, settings.IntProperty);
         }
 
+        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer)),
+            InlineData(true),
+            InlineData(false)]
+        [ActiveIssue("dotnet/corefx #18832", TargetFrameworkMonikers.NetFramework)]
```
Comment (ericstj) [inline src/System.Configuration.ConfigurationManager/tests/System/Configuration/ApplicationSettingsBaseTests.cs]: I don’t think this should be failing with NotSupported.  I’d expect this to just forward to the desktop implementation.  The linked issue looks like it’s lacking a good root cause analysis.

File: src/System.Configuration.ConfigurationManager/tests/System/Configuration/ApplicationSettingsBaseTests.cs [hunk 4]
```diff
@@ -103,12 +107,59 @@ public void GetSetIntProperty_SimpleSettings_Ok(bool isSynchronized)
...
+        [ActiveIssue("https://github.com/dotnet/coreclr/issues/35668", TestPlatforms.AnyUnix)]
```
Comment (ericstj) [inline src/System.Configuration.ConfigurationManager/tests/System/Configuration/ApplicationSettingsBaseTests.cs]: This link gives a 404.  It looks like it should be corefx.

**Pull request merged**